src/libraries/System.Net.Quic/src/System/Net/Quic/QuicStream.cs

```diff
@@ -351,29 +351,32 @@ public override ValueTask WriteAsync(ReadOnlyMemory<byte> buffer, CancellationTo
     /// <param name="completeWrites">Notifies the peer about gracefully closing the write side, i.e.: sends FIN flag with the data.</param>
     public ValueTask WriteAsync(ReadOnlyMemory<byte> buffer, bool completeWrites, CancellationToken cancellationToken = default)
     {
-        ObjectDisposedException.ThrowIf(_disposed == 1, this);
+        if (_disposed == 1)
+        {
+            return ValueTask.FromException(new ObjectDisposedException(nameof(QuicStream)));
```

Comment:
```suggestion
            return ValueTask.FromException(ExceptionDispatchInfo.SetCurrentStackTrace(new ObjectDisposedException(nameof(QuicStream))));
```

Applies for the others as well.

Reply:
Was wondering about this, we don't use `SetCurrentStackTrace` consistently everywhere (e.g. we use it for Connection events but not for Stream events). Should we use it everywhere?

Reply:
`(Ressetable)ValueTaskSource` does it for us. So we have to use it manually only for cases like this. But if we have some places missing, we should fix them.

Reply:
That explains it. I don't think we have any places missing it then.